activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -118,13 +150,8 @@ def sum(identity = nil)
   class Array
     alias :orig_sum :sum
 
-    def sum(init = nil, &block) #:nodoc:
-      if init.is_a?(Numeric) || first.is_a?(Numeric)
-        init ||= 0
-        orig_sum(init, &block)
-      else
-        super
-      end
+    def sum(identity = nil, &block) #:nodoc:
+      super
```

Comment:
Ruby has separate [Array#sum](https://github.com/ruby/ruby/blob/trunk/array.c#L5658-L5700) and [Enumerable#sum](https://github.com/ruby/ruby/blob/trunk/enum.c#L3715-L3744) implementations, so we'll want to continue using `orig_sum` for the optimized Array implementation but calling `super` to get our compatible `#sum` behavior.